packages/hash/subgraph/src/stdlib/edge/entity-type.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  BaseUri,
+  extractBaseUri,
+  extractVersion,
+} from "@blockprotocol/type-system";
+
+import { isConstrainsPropertiesOnEdge } from "../../types/edge/outward-edge-alias";
+import { OntologyTypeEditionId, VersionedUri } from "../../types/identifier";
+import { Subgraph } from "../../types/subgraph";
+
+/**
+ * Gets identifiers for all `PropertyType`s referenced within a given `EntityType` schema by searching for
+ * "ConstrainsPropertiesOn" `Edge`s from the respective `Vertex` within a `Subgraph`.
+ *
+ * @param subgraph {Subgraph} - The `Subgraph` containing the type tree of the `EntityType`
+ * @param entityTypeId {OntologyTypeEditionId | VersionedUri} - The identifier of the `EntityType` to search for
+ * @returns {OntologyTypeEditionId[]} - The identifiers of the `PropertyType`s referenced from the `EntityType`
+ */
+export const getPropertyTypesReferencedByEntityType = (
+  subgraph: Subgraph,
+  entityTypeId: OntologyTypeEditionId | VersionedUri,
+): OntologyTypeEditionId[] => {
```

Comment:
I'm unsure if we want to return `PropertyType[]` or just the identifiers. If we do return identifiers I'm also unsure if we want to return `VersionedUri[]` or `OntologyTypeEditionId[]` as here. 

`OntologyTypeEditionId[]` is more efficient for future lookups as you can just do `subgraph.vertices[editionId.baseId][editionId.version]` but `VersionedUri` requires you calling `extract` again.

Returning `PropertyType[]` prematurely also doesn't feel ideal because we are assuming they need the whole object, and that involves more copying of data and more lookups